src/app/clusters/time-synchronization-server/time-synchronization-server.h

```diff
@@ -0,0 +1,170 @@
+            mDefaultNtp.SetNonNull(MutableByteSpan(mDefaultNtpBuf, dntp.Value().size()));
+            return mTimeSyncDataProvider.StoreDefaultNtp(mDefaultNtp.Value());
+        }
+        else
+        {
+            mDefaultNtp.SetNull();
+            return mTimeSyncDataProvider.ClearDefaultNtp();
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetTimeZone(DataModel::DecodableList<TimeSynchronization::Structs::TimeZoneStruct::Type> tz)
+    {
+        auto mTzL = mTimeZoneList.begin();
+        auto tzL  = tz.begin();
+        size_t i  = 0;
+
+        while (tzL.Next())
+        {
+            mTzL[i].offset  = tzL.GetValue().offset;
+            mTzL[i].validAt = tzL.GetValue().validAt;
+            if (tzL.GetValue().name.HasValue())
+            {
+                const char * buf = tzL.GetValue().name.Value().data();
+                size_t len       = tzL.GetValue().name.Value().size();
+                Platform::CopyString(mNames[i].name, chip::CharSpan(buf, len));
+            }
+            i++;
+        }
+
+        return mTimeSyncDataProvider.StoreTimeZone(mTimeZoneList);
+    }
+
+    CHIP_ERROR SetDSTOffset(DataModel::DecodableList<TimeSynchronization::Structs::DSTOffsetStruct::Type> dst)
+    {
+        auto mDstL = mDstOffList.begin();
+        auto dstL  = dst.begin();
+        size_t i   = 0;
+
+        while (dstL.Next())
+        {
+            mDstL[i] = dstL.GetValue();
+            i++;
+        }
+
+        return mTimeSyncDataProvider.StoreDSTOffset(mDstOffList);
+    }
+
+    CHIP_ERROR ClearDSTOffset()
+    {
+        for (size_t i = 0; i < CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE; i++)
+        {
+            mDst[i] = { 0 };
+        }
+
+        return mTimeSyncDataProvider.ClearDSTOffset();
+    }
+
+    DataModel::Nullable<TimeSynchronization::Structs::TrustedTimeSourceStruct::Type> & GetTrustedTimeSource(void)
+    {
+        return mTrustedTimeSource;
+    }
+    DataModel::Nullable<chip::MutableByteSpan> & GetDefaultNtp(void) { return mDefaultNtp; }
+    DataModel::List<TimeSynchronization::Structs::TimeZoneStruct::Type> & GetTimeZone(void) { return mTimeZoneList; }
+    DataModel::List<TimeSynchronization::Structs::DSTOffsetStruct::Type> & GetDSTOffset(void) { return mDstOffList; }
+
+    void ScheduleDelayedAction(System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState);
+
+private:
+    DataModel::Nullable<TimeSynchronization::Structs::TrustedTimeSourceStruct::Type> mTrustedTimeSource;
+    DataModel::Nullable<chip::MutableByteSpan> mDefaultNtp;
+    DataModel::List<TimeSynchronization::Structs::TimeZoneStruct::Type> mTimeZoneList =
+        DataModel::List<TimeSynchronization::Structs::TimeZoneStruct::Type>(mTz, CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE);
+    DataModel::List<TimeSynchronization::Structs::DSTOffsetStruct::Type> mDstOffList =
+        DataModel::List<TimeSynchronization::Structs::DSTOffsetStruct::Type>(mDst, CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE);
```

Comment:
Just leave off the size if you want it to be the "entire size of the array"....  Here and for mTimeZoneList above.